client_side/.infrastructure/setup.sh

```diff
@@ -0,0 +1,238 @@
+################################################################################
+
+# Makes sure that all the scripts are executable
+chmod +x "${INFRA_DIR}"/*.sh
+chmod +x "${INFRA_DIR}"/*.py
+
+# Establish infrastructure variables and functions
+source "${INFRA_DIR}"/infrastructure.sh
+touch "${INFRA_DIR}"/.{task_code,curr_task,treatment,task_order,command}
+
+# Initalize variable with default values
+echo "start task" > "${INFRA_DIR}/.command"
+
+status="incomplete"
+time_elapsed=0
+
+# If a curr_task file already exists, it means we are trying to resume the
+# experiment
+if [[ -e "${INFRA_DIR}/.curr_task" ]]; then
+  curr_task=$(cat "${INFRA_DIR}/.curr_task")
+else
+  curr_task=1
+  echo "${curr_task}" > "${INFRA_DIR}/.curr_task"
+fi
+
+# Determine the task order based on a truncated md5sum hash of the username.
+# It is two two-character codes.  In each two-character code, the letter
+# T/N is for Tellina/NoTellina, and the number indicates the task_set used.
+case $(echo $((0x$(md5sum <<<${USER_NAME} | cut -c1) % 4))) in
+  0)
+    echo "T1N2" > "${INFRA_DIR}/.task_order"
+    echo "T" > "${INFRA_DIR}/.treatment"
+    task_set=1
+    ;;
+  1)
+    echo "T2N1" > "${INFRA_DIR}/.task_order"
+    echo "T" > "${INFRA_DIR}/.treatment"
+    task_set=2
+    ;;
+  2)
+    echo "N1T2" > "${INFRA_DIR}/.task_order"
+    echo "N" > "${INFRA_DIR}/.treatment"
+    task_set=1
+    ;;
+  3)
+    echo "N2T1" > "${INFRA_DIR}/.task_order"
+    echo "N" > "${INFRA_DIR}/.treatment"
+    task_set=2
+    ;;
+esac
+
+# Determines the task code w.r.t current_task and task_set
+echo $(get_task_code) > "${INFRA_DIR}/.task_code"
+
+# Create user meta-commands
+alias reset="touch ${INFRA_DIR}/.reset"
+alias task="touch ${INFRA_DIR}/.task"
+alias abandon="touch ${INFRA_DIR}/.abandon"
+alias helpme="touch ${INFRA_DIR}/.helpme"
+
+################################################################################
+#                                  BASH PREEXEC                                #
+################################################################################
+
+# Install Bash preexec.
+source "${INFRA_DIR}"/bash-preexec.sh
+
+# Executed before the user-entered command is executed
+# Saves the most recent command into the .command file
+# If the user enters an "empty" command, then the .command file does not change
+preexec_func() {
+  echo "$1" > "${INFRA_DIR}/.command"
+}
+
+# Executed after the user-entered command is executed
+# - Only one of these cases can happen
+# 1. Check if user has run out of time:
+#    - `time_elapsed=$SECONDS` is less than some time limit constant.
+#    - The check will happen after the command is executed.
+#    - If the user ran out of time, `status="timeout"`,
+#      `time_elapsed=$TIME_LIMIT`.
+# 2. Handle user meta-command:
+#    - Output verification will not be performed on these commands.
+#    - The check is done by looking for the existence of the file
```

Comment:
I think it would be cleaner to put the functionality in the aliases for `abandon`, etc instead.  That would dispense with the need to create and delete these marker files.
You would still have a marker file for .noverify (meaning that no output verification should be done, which is relevant to all 4 special commands and to starting the task), but this is simpler than having 5 `if`s in this file.

Reply:
Commit 8af6cc0 has a solution that is up for consideration.